backend/src/main/java/com/team4/backend/controller/StudentController.java

```diff
@@ -30,4 +34,15 @@ public StudentController(StudentService studentService) {
         //TODO add a non-handled exception to make sure it returns 500 and not 409
     }
 
+    @PatchMapping("/updateStudentState")
+    @PreAuthorize("hasAuthority('STUDENT')")
+    public Mono<ResponseEntity<String>> updateStudentState(Principal principal) {
+        return studentService.updateStudentState(UserSessionService.getLoggedUserEmail(principal), StudentState.INTERNSHIP_FOUND)
+                .flatMap(s -> Mono.just(ResponseEntity.status(HttpStatus.NO_CONTENT).body("")))
```

Comment:
you can filter exceptions with onErrorResume by putting the ExceptionName.class as the first argument.
So you could have 2 onErrorResume instead